### ISSUE TITLE
Support room versioning

### ISSUE THREAD

**Issue by KitsuneRal** [quotient-im/libQuotient#233]
See https://github.com/matrix-org/matrix-doc/issues/1425 and https://github.com/matrix-org/matrix-doc/pull/1516. Room upgrading is out of scope of this issue.

**Comment by KitsuneRal** [comment]
The support in this case boils down to the following:
- [ ] Create the list of supported API versions within the library (and maintain it from now on)
- [ ] Add `room_version` to `CreateRoomJob` (this should come automatically due to https://github.com/matrix-org/matrix-doc/pull/1516) and `Connection::createRoom`; the latter should correspond the omitted room version with the highest one supported by the library _and_ the server (I suspect a gap in CS API here for now).
- [ ] Avoid processing rooms with unsupported versions (the rooms should be added to the list of connections and some basic state events like name and canonical alias should be parsed but no further processing should be done; depends on #234).
- [ ] Propagate to clients an error from joining a room the version of which the server doesn't support (`M_UNSUPPORTED_ROOM_VERSION`).

**Comment by KitsuneRal** [comment]
After some time in the field with the mentioned above two of the four items implemented, it looks like the other two are not really helping anything.
